lib/src/main/java/com/hootsuite/nachos/NachoTextView.java

```diff
@@ -727,6 +727,128 @@ public void setTextWithChips(@Nullable List<ChipInfo> chips) {
         endUnwatchedTextChange();
     }
 
+    /**
+     *
+     * @return String of unchipped text that AutoCompleteTextView sees
+     */
+
+    public String getTextForSuggestion() {
+        if (getChipTokenizer() != null) {
+            int end = getSelectionEnd();
+            Editable editable = getText();
+            int start = getChipTokenizer().findTokenStart(editable, end);
+            if (start != -1 && end != -1) {
+                if (end < start) {
+                    start = end;
+                }
+                return editable.subSequence(start, end).toString();
+            }
+        }
+        return "";
+    }
+
+    /**
+     *
+     * @param text to check if exists with Chip
+     * @return true if an existing chip contains text and null object
+     */
+    public boolean isAlreadyAdded(CharSequence text) {
+        return isAlreadyAdded(text, null);
+    }
+
+    /**
+     *
+     * @param text to check if exists with Chip
+     * @param object to check if exists with Chip
+     * @return true if an existing chip contains text and object
+     */
+    public boolean isAlreadyAdded(CharSequence text, Object object) {
+        List<Chip> chips = getAllChips();
+        for (Chip chip: chips) {
+            CharSequence existingText = chip.getText();
+            Object existingObject = chip.getData();
+            if (text.equals(existingText) &&
+                    (object == null && existingObject == null) || (object != null && object.equals(existingObject))) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     *
+     * @param text creates a chip with specified text
+     */
+    public void insertChip(CharSequence text) {
+        insertChip(text, null);
+    }
+
+    /**
+     *
+     * @param text creates a chip with specified text
+     * @param object creates a chip with specified object
+     */
+    public void insertChip(CharSequence text, Object object) {
+        if (mChipTokenizer != null) {
+            clearComposingText();
+
+            int end = getSelectionEnd();
```

Comment:
This is interesting, you want to replace a token based on the `getSelection`. I think this is when you have a suggestion and when the user taps on a suggestion, you want to replace the token with with the chip. I see. 

Originally, I thought this method was to just insert a chip at the end of the `NachoTextView` and not by cursor position. Some thoughts

- I think an `insertChip()` method makes sense for just adding a chip at current cursor position or at the end of the `NachoTextView` as the name of the method seems to indicate that. 
- I seem to notice a them to work with Suggestions and AutoComplete, you can follow a similar them with `AutoCompleteTextView` and create a Suggestion supported `NachoTextView` to provide this `insertChip` functionality to replace a token (by cursor position) with a chip.